pkg/tests/types.go

```diff
@@ -0,0 +1,24 @@
+package tests
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// MockEventRecorder mocks EventRecorder, which knows how to record events on behalf of an EventSource.
+type MockEventRecorder struct{}
```

Comment:
How about using `record.NewFakeRecorder` ?
https://github.com/Azure/application-gateway-kubernetes-ingress/blob/c9cd1446aecf5d827ca9f967d51c64fd43aa180d/pkg/appgw/appgw_test.go#L401

Reply:
This is great - always happy to reuse great stuff!